### ISSUE TITLE
Access Violation using DirectMessage for SCI_AUTOCSETFILLUPS

### ISSUE THREAD

**Issue by ODGMike** [jacobslusser/ScintillaNET#70]
I know using DirectMessage is at your own risk but would appreciate it if you could take a look.  I was successfully using DirectMessage for SCI_AUTOCSETFILLUPS with release 3.4 but after updating to 3.5 I get an access violation.

An unhandled exception of type 'System.AccessViolationException' occurred in ScintillaNET.dll

**Comment by jacobslusser** [comment]
No changes have been made in the area of `DirectMessage`. I would also conclude that if there was a problem it would be evident in many, many places because all of ScintillaNET API calls funnel down to `DirectMessage`.

So with that in mind, I'm wondering if you have an issue in your calling code. I see access violations when memory is not properly pinned. Could you send me some sample code?

**Comment by jacobslusser** [comment]
So I traced through the Scintilla code and much to my surprise, it doesn't appear that Scintilla is making a copy of the string passed to SCI_AUTOCSETFILLUPS. That means that any string of fillup characters you pass in must remain in the same place in memory for the life of the control. Managed runtimes like .NET don't make those kind of guarantees. You would need to explicitly write code that would either pin the managed string or keep it in an unmanaged location outside of the garbage collector's reach.

If that wasn't something you were explicitly doing in your application code you were simply lucky and the `AccessViolationException` just finally reared its ugly head around the time you upgraded to v3.5.

Since writing that kind of code can be difficult and I needed to add support for SCI_AUTOCSETFILLUPS anyway, I went ahead and did so in commit ec730e16092e78c002ffb14ec352ed2708fb4be6.

Let me know if are still having any issues.

P.S. Hence the warnings about using `DirectMessage`. ;)
